CONTRIBUTING.md

```diff
@@ -1,3 +1,540 @@
+
+To test this, you will need to clone the PySyft library, install the dependencies and link the PySyft src as we have done with SyMPC. For more detailed information, please visit the [PySyft contributin guide](https://github.com/OpenMined/PySyft/blob/dev/packages/syft/CONTRIBUTING.md).
+
+As a little summary, you need to:
+
+```bash
+$ git clone https://github.com/OpenMined/PySyft.git
+$ cd PySyft/packages
+$ pip install -r 'requirements.txt'
+$ pip install -e syft
+```
+
+This steps are here as a guide. Please, check the [PySyft contributin guide](https://github.com/OpenMined/PySyft/blob/dev/packages/syft/CONTRIBUTING.md) for the detailed steps.
+
+Now, you are prepared to face this kind of changes. A normal change in SyMPC looks like:
+
+1. Change SyMPC code
+2. Run the tests
+
+A change that implies both libraries looks like:
+
+1. Change SyMPC code
+2. Change PySyft code
+3. Install SyMPC with `pip install -e .`
+4. Install PySyft with `pip install -e syft`
+5. Run SyMPC tests
+6. Run PySyft tests
+
+## Code Quality
+
+### Formatting and Linting
+
+We use several tools to keep our codebase high quality.
+
+- black
+- flake8
+- isort
+
+### Tests and CI
+
+When you push your code it will run through a series of GitHub Actions which will ensure that the code meets our minimum standards of code quality before a Pull Request can be reviewed and approved.
+
+To make sure your code will pass these CI checks before you push you should use the pre-commit hooks and run tests locally.
+
+We aim to have a 100% test coverage, and the GitHub Actions CI will fail if the coverage is below a certain value. You can evaluate your coverage using the following commands.
+
+```
+$ pytest -m fast -n auto
+```
+
+### Writing Test Cases
+
+Always make sure to create the necessary tests and keep test coverage at 100%. You can always ask for help in slack or via GitHub if you don't feel confident about your tests.
+
+### Documentation and Code Style Guide
+
+To ensure code quality and make sure other people can understand your changes, you have to document your code. For documentation, we are using the Google Python Style Rules which can be found [here](https://github.com/google/styleguide/blob/gh-pages/pyguide.md). A well-written example can we viewed [here](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
+
+Your documentation should not describe the obvious, but explain what's the intention behind the code and how you tried to realize your intention.
+
+You should also document non-self-explanatory code fragments e.g. complicated for-loops. Again please do not just describe what each line is doing but also explain the idea behind the code fragment and why you decided to use that exact solution.
+
+You can check documentation style errors with:
+
+```bash
+# Check if the description matches a function signature
+$ darglint .
+# Checks compliance with python docstrings conventions
+$ pydocstyle .
```

Comment:
@jmaunon we do this in the `pre-commit hooks`, right?

Reply:
If I am not wrong, I get the information from this file.

![image](https://user-images.githubusercontent.com/20444151/118391545-0a769e80-b635-11eb-990c-e4b38b9fb334.png)